coalib/results/TextRange.py

```diff
@@ -89,7 +89,20 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        return (self.is_behind(self.start, other.end) and
+                self.is_behind(other.start, self.end))
+
+    @staticmethod
+    def is_behind(first_position, second_position):
+        if first_position.line is None or second_position.line is None:
+            return True
+        if first_position.line == second_position.line:
+            return (True
+                    if None in (first_position.column, second_position.column)
```

Comment:
you should expand this one too to `if first_position.column is None or ...`, I tested this with `timeit`:

```python
>>> from timeit import *
>>> def in_approach(x, y):
...  if None in (x, y):
...   return 3
...  else:
...   return 4
... 
>>> def direct_approach(x, y):
...  if x is None or y is None:
...   return 3
...  else:
...   return 4
... 
>>> data = []
>>> for i in range(100):
...  if i % 3 == 0:
...   data.append((None, i))
...  elif i % 7 == 0:
...   data.append((i, None))
...  else:
...   data.append((i, i))
... 
>>> def IN():
...  for x, y in data:
...   in_approach(x, y)
... 
>>> def DIRECT():
...  for x, y in data:
...   direct_approach(x, y)
... 
>>> T_IN = Timer(IN)
>>> T_DIRECT = Timer(DIRECT)
>>> count = 100000
>>> r = (T_IN.timeit(count), T_DIRECT.timeit(count))
>>> r
(1.7751383920000308, 1.2874540749999142)
>>> r[1] / r[0]
0.698429337908
```
You see it's about 30% faster, this is quite much^^